api/tests/unit/environments/identities/test_unit_identities_views.py

```diff
@@ -1166,10 +1179,68 @@ def test_post_identities__transient__no_persistence(
 
     # Then
     assert response.status_code == status.HTTP_200_OK
-    assert not Identity.objects.filter(identifier=identifier).exists()
+    assert not Identity.objects.exists()
     assert not Trait.objects.filter(trait_key=trait_key).exists()
 
 
+@pytest.mark.parametrize(
+    "trait_transiency_data",
+    [
+        pytest.param({"transient": True}, id="trait-transient-true"),
+        pytest.param({"transient": False}, id="trait-transient-false"),
+        pytest.param({}, id="trait-default"),
+    ],
+)
+def test_post_identities__existing__transient__no_persistence(
+    environment: Environment,
+    identity: Identity,
+    trait: Trait,
+    identity_featurestate: FeatureState,
+    api_client: APIClient,
+    trait_transiency_data: dict[str, Any],
+) -> None:
+    # Given
+    feature_state_value = "identity override"
+    identity_featurestate.feature_state_value.string_value = feature_state_value
+    identity_featurestate.feature_state_value.save()
+
+    trait_key = "trait_key"
+
+    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment.api_key)
+    url = reverse("api-v1:sdk-identities")
+    data = {
+        "identifier": identity.identifier,
+        "transient": True,
+        "traits": [
+            {"trait_key": trait_key, "trait_value": "bar", **trait_transiency_data}
+        ],
+    }
+
+    # When
+    response = api_client.post(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_200_OK
+    response_json = response.json()
+
+    assert response_json["flags"][0]["feature_state_value"] == feature_state_value
+    assert response_json["traits"][0]["trait_key"] == trait.trait_key
+    assert not response_json["traits"][0].get("transient")
+
+    assert response_json["traits"][1]["trait_key"] == trait_key
+    assert response_json["traits"][1]["transient"]
```

Comment:
This confused me for a while but I think this is because, when the identity is transient, we mark all traits as transient too? I think it might be worth adding a comment here to confirm that.

Reply:
Added more comments to the test, hopefully it's clearer now.